### PR TITLE
Configure and run mypy for type checking

### LINKED ISSUE #103: Add mypy type checker
*This is a good first issue for new contributors to take on, if you have any questions, please ask on the task or in our [Gitter room](https://gitter.im/freedomofpress/securedrop)!*

We should use the [`mypy`](https://mypy.readthedocs.io/en/stable/) type checker in this repository. You can probably copy most of the configuration from https://github.com/freedomofpress/securedrop-client. It would be good to fix simple issues in the same PR, but if some types are complicated, it's fine to mark them with `# type: ignore` to suppress the mypy warning for now so it can be cleaned up at a later time.

### PR DESCRIPTION
**Pull request by legoktm**
Fixes #103.

Split into 3 commits (each commit message has more detail):
1. Configure mypy and add to CI
2. Fix all the things that I found straightforward/easy
3. Suppress and leave FIXMEs for things I wasn't sure about.

The last commit will need someone more familiar with the codebase (@rocodes?) to resolve some of the issues. Right now I threw in some exceptions so it probably shouldn't be merged as is, but I can switch them to just pure suppression of mypy errors if we don't have a good solution or time to properly fix them.

## Test plan
* [ ] CI passes
* [ ] `make mypy` passes locally

### CODE REVIEW

**Comment by rocodes** [comment]
Thanks for getting the ball rolling, @legoktm, and sorry for all the errors you encountered. You got the worst of it looking at a couple files that aren't being used yet, but is in place to support future work in the client. I'll push a couple fixes to this branch.

File: requirements/dev-bookworm-requirements.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #
-# This file is autogenerated by pip-compile with Python 3.10
+# This file is autogenerated by pip-compile with Python 3.9
```
Comment (rocodes) [inline requirements/dev-bookworm-requirements.txt:2]: Shouldn't this be 3.10?

Comment (legoktm) [reply requirements/dev-bookworm-requirements.txt:2]: Yes...

Comment (legoktm) [reply requirements/dev-bookworm-requirements.txt]: OK, that just made it worse. I need to double check how we fixed this in sd-client.

**Comment by rocodes** [comment]
Fixes for the non-CI issues: (https://github.com/rocodes/securedrop-export/commits/mypy-fixes aka https://github.com/rocodes/securedrop-export/commit/828a8f653687369c613f6a846183dd91c3a38460 + https://github.com/rocodes/securedrop-export/commit/828a8f653687369c613f6a846183dd91c3a38460)

File: requirements/dev-bookworm-requirements.txt (hunk 1 above)
Comment (legoktm) [reply requirements/dev-bookworm-requirements.txt]: cherry-picked https://github.com/freedomofpress/securedrop-client/commit/37292714b97b1147801de408d41889b9a5d63205 to fix it!

**Comment by legoktm** [comment]
I think this is ready to be re-reviewed + get merged now!

File: securedrop_export/disk/cli.py [hunk 2]
```diff
@@ -324,22 +324,24 @@ def _mount_at_mountpoint(self, volume: Volume, mountpoint: str) -> Volume:
...
 
-        return volume
+        return MountedVolume.from_volume(volume, mountpoint)
 
     def write_data_to_device(
-        self, submission_tmpdir: str, submission_target_dirname: str, device: Volume
+        self,
+        submission_tmpdir: str,
+        submission_target_dirname: str,
+        device: MountedVolume,
```
Comment (gonzalo-bulnes) [inline securedrop_export/disk/cli.py:337]: Nice!

File: securedrop_export/disk/legacy_service.py [hunk 3]
```diff
@@ -87,16 +95,18 @@ def export(self):
                 if self.cli.is_luks_volume(device):
                     volume = self.cli.get_luks_volume(device)
                     logger.info("Check if writable")
-                    if not volume.writable:
+                    if not isinstance(volume, MountedVolume):
```
[the remark this answers is not in the document]
Comment (gonzalo-bulnes) [reply securedrop_export/disk/legacy_service.py:98]: Got the change in https://github.com/freedomofpress/securedrop-export/pull/120/commits/d57c18239d1950c55a085896e78e610536a6106d

**Review by gonzalo-bulnes — APPROVED** [review]
All this looks great to me :+1:

**Pull request merged**